src/Ztm.WebApi/TransactionConfirmationWatchers/TransactionConfirmationWatcher.cs

```diff
@@ -392,13 +392,15 @@ async Task<IEnumerable<Rule>> ITransactionConfirmationWatcherHandler<Rule>.Creat
         {
             if (reason.HasFlag(WatchRemoveReason.Completed))
             {
-                await this.watchRepository.UpdateStatusAsync(watch.Id, TransactionConfirmationWatchingWatchStatus.Success, cancellationToken);
+                await this.watchRepository.UpdateStatusAsync(watch.Id, WatchStatus.Success, cancellationToken);
             }
             else if (reason.HasFlag(WatchRemoveReason.BlockRemoved))
             {
-                await this.watchRepository.UpdateStatusAsync(watch.Id, TransactionConfirmationWatchingWatchStatus.Rejected, cancellationToken);
+                await this.watchRepository.UpdateStatusAsync(watch.Id, WatchStatus.Rejected, cancellationToken);
                 await SetupTimerAsync(watch.Context);
             }
+
+            await this.ruleRepository.ClearCurrentWatchIdAsync(watch.Context.Id, CancellationToken.None);
```

Comment:
This one should come before a call to `SetupTimerAsync` due to rule state should be ready to start timer.